File: Java/LeetCode/Sort Even and Odd Indices Independently.java
// https://leetcode.com/problems/sort-even-and-odd-indices-independently/

class Solution {

    public int[] sortEvenOdd(int[] nums) {
        List<Integer> oddIndexNumbers = new ArrayList<>();
        List<Integer> evenIndexNumbers = new ArrayList<>();

        for (int i = 0; i < nums.length; ++i) {
            if (i % 2 == 0) {
                evenIndexNumbers.add(nums[i]);
            } else {
                oddIndexNumbers.add(nums[i]);
            }
        }

        Collections.sort(evenIndexNumbers);
        Collections.sort(oddIndexNumbers, Collections.reverseOrder());

        int[] sortedArr = new int[nums.length];
        for (int i = 0, j = 0, k = 0; i < nums.length; ++i) {
            sortedArr[i] = (i % 2 == 0) ? evenIndexNumbers.get(j++) : oddIndexNumbers.get(k++);
        }

        return sortedArr;
    }

}

File: Kotlin/LeetCode/Sort Even and Odd Indices Independently.kt
// https://leetcode.com/problems/sort-even-and-odd-indices-independently/

class Solution {

    fun sortEvenOdd(nums: IntArray): IntArray {
        var oddIndexNumbers = arrayListOf<Int>();
        var evenIndexNumbers = arrayListOf<Int>();

        for (i in 0 until nums.size) {
            if (i % 2 == 0) {
                evenIndexNumbers.add(nums[i]);
            } else {
                oddIndexNumbers.add(nums[i]);
            }
        }

        evenIndexNumbers.sort();
        oddIndexNumbers.sortDescending();

        var sortedArr = IntArray(nums.size)
        var j = 0
        var k = 0
        for (i in 0 until nums.size) {
            sortedArr[i] = if (i % 2 == 0) evenIndexNumbers.get(j++) else oddIndexNumbers.get(k++)
        }

        return sortedArr;
    }

}
